### ISSUE TITLE
Found a fatal bug

### ISSUE THREAD

**Issue by CockyCat** [decred/dcrd#1167]
When I used getnewaddress, reported this error: deriving additional addresses violates the unused address gap limit

How can I break through the limitations of generating addresses?

solution: use `dcrctl --wallet getnewaddress default ignore`

Issue:  

When using the address generated by dcrctl --wallet getnewaddress default ignore and then sending transactions to these addresses, these addresses do not receive dcr. But they can be found through the block browser and the sender does deduct Dcr

Please pay attention to this issue, which means that DCR is lost?

**Comment by chappjc** [comment]
Are you aware of the function of the address gap limit?  The implications of using the `ignore` option?

Also, it's likely the issue you're having is related to dcrwallet, not dcrd.

**Comment by davecgh** [comment]
As stated, you filed this on the wrong repository as addresses are handled by wallet and as such should be filed against that repository.  That said, there is no bug here, rather you are using an option without understanding the implications of it.

There is a reason an error is returned if you would exceed the configured gap limit by default, and it's precisely because it requires you to make a choice between the different behaviors according to your use case which can lead to you needing to manually configure options, as is the case here.  If you're going to use the ignore option, then you must later manually start wallet with a gap limit large enough to cover the largest run of unused addresses you obtained in order to be able to detect and account for the transactions involving those.

```
$ dcrwallet -h
...
     --addridxscanlen:                         The width of the scan for last
                                               used addresses on wallet restore
                                               and start up (default: 20)
...
```

In other words, you need to start `dcrwallet` with the `--addridxscanlen` option set to a value high to cover the largest run of unused addresses you obtained .

**Comment by CockyCat** [comment]
I'm set addridxscanlen=10000 in dcrwallet.conf,  Then restart the dcrd and dcrwallet processes, but the transaction that was sent has not been received yet
